acceptance/setup/aio/pre-suite/010_Install.rb

```diff
@@ -35,8 +38,20 @@
 
 step "Install puppetserver..." do
   if ENV['SERVER_VERSION']
-    install_puppetlabs_dev_repo(master, 'puppetserver', ENV['SERVER_VERSION'])
-    install_puppetlabs_dev_repo(master, 'puppet-agent', ENV['SHA'])
+    ENV['AGENT_DOWNLOAD_URL'] ||= DEFAULT_INTERNAL_DOWNLOAD_URL
+    ENV['SERVER_DOWNLOAD_URL'] ||= DEFAULT_NIGHTLIES_DOWNLOAD_URL
+    # We can only use a SHAs of "latest" if we're downloading from nightlies, as
+    # our other options, such as 'builds.delivery.puppetlabs.net', do not have
+    # "latest" directories.
+    if ENV['SHA'] == 'latest' and ENV['AGENT_DOWNLOAD_URL'] != DEFAULT_NIGHTLIES_DOWNLOAD_URL
+      fail "Can only use ENV['SHA'] == 'latest' if ENV['AGENT_DOWNLOAD_URL'] == '#{DEFAULT_NIGHTLIES_DOWNLOAD_URL}'"
+    end
+
+    if ENV['SERVER_VERSION'] == 'latest' and ENV['SERVER_DOWNLOAD_URL'] != DEFAULT_NIGHTLIES_DOWNLOAD_URL
+      fail "Can only use ENV['SERVER_VERSION'] == 'latest' if ENV['SERVER_DOWNLOAD_URL'] == '#{DEFAULT_NIGHTLIES_DOWNLOAD_URL}'"
+    end
+    install_puppetlabs_dev_repo(master, 'puppetserver', ENV['SERVER_VERSION'], nil, :dev_builds_url => ENV['SERVER_DOWNLOAD_URL'])
+    install_puppetlabs_dev_repo(master, 'puppet-agent', ENV['SHA'], nil, :dev_builds_url => ENV['AGENT_DOWNLOAD_URL'])
     master.install_package('puppetserver')
   else
     # beaker can't install puppetserver from nightlies (BKR-673)
```

Comment:
The beaker issue that prevented nightlies from working has been fixed, so I think we could simplify all of this logic to use `install_puppetlabs_dev_repo` when testing against a specific puppetserver version or latest:

```diff
diff --git a/acceptance/setup/aio/pre-suite/010_Install.rb b/acceptance/setup/aio/pre-suite/010_Install.rb
index 09a5901..4e71ab1 100644
--- a/acceptance/setup/aio/pre-suite/010_Install.rb
+++ b/acceptance/setup/aio/pre-suite/010_Install.rb
@@ -34,17 +34,17 @@ MASTER_PACKAGES = {
 }
 
 step "Install puppetserver..." do
-  if ENV['SERVER_VERSION']
-    install_puppetlabs_dev_repo(master, 'puppetserver', ENV['SERVER_VERSION'])
-    install_puppetlabs_dev_repo(master, 'puppet-agent', ENV['SHA'])
-    master.install_package('puppetserver')
+  if ENV['SERVER_VERSION'].nil? || ENV['SERVER_VERSION'] == 'latest'
+    server_version = 'latest'
+    server_download_url = "http://nightlies.puppet.com"
   else
-    # beaker can't install puppetserver from nightlies (BKR-673)
-    repo_configs_dir = 'repo-configs'
-    install_repos_on(master, 'puppetserver', 'nightly', repo_configs_dir)
-    install_repos_on(master, 'puppet-agent', ENV['SHA'], repo_configs_dir)
-    install_packages_on(master, MASTER_PACKAGES)
+    server_version = ENV['SERVER_VERSION']
+    server_download_url = "http://builds.delivery.puppetlabs.net"
   end
+
+  install_puppetlabs_dev_repo(master, 'puppetserver', server_version, nil, :dev_builds_url => server_download_url)
+  install_puppetlabs_dev_repo(master, 'puppet-agent', ENV['SHA'])
+  master.install_package('puppetserver')
 end
 
 # make sure install is sane, beaker has already added puppet and ruby
```

So then the server version can be specified as:

```
$ bundle exec rake ci:test:aio TEST_TARGET=redhat6-64a SHA=1.10.1 SUITE_VERSION=1.10.1 SERVER_VERSION=2.7.1
...
     puppetserver       noarch 2.7.1-1.el7             pl-puppetserver-2.7.1   33 M
```

vs

```
$ rake ci:test:aio TEST_TARGET=redhat6-64a SHA=1.10.1 SUITE_VERSION=1.10.1 SERVER_VERSION=latest
...
 puppetserver       noarch 5.0.0.master-0.1SNAPSHOT.2017.05.24T1548.el7
```